utils.py

```diff
@@ -759,96 +759,210 @@ def melt_kt_grouped(kt, columns_to_melt, value_column_names, key_column_name='va
-                   i_col='i',
-                   j_col='j',
-                   kin_col='kin',
-                   k2_col='k2',
-                   sex_s_col='s',
-                   sex_sex_col='sex',
-                   first_degree_threshold=(0.23, 0.27),  # TODO: Look at distribution
-                   duplicate_threshold=0.04,  # TODO: Look at distribution
-                   k2_parent_offspring_threshold=0.1  # TODO: Look at distribution
-                   ):
-
-
-    def get_samples_relatives(sample_pairs):
-        samples_rel = defaultdict(set)
-        for row in sample_pairs:
-            samples_rel[row[i_col]].add(row[j_col])
-            samples_rel[row[j_col]].add(row[i_col])
-        return samples_rel
-
-
-    def prune_relatives(relatives):
-        pruned = set()
-        for s, s_rel in relatives:
-            if s not in pruned:
-                pruned.update(s_rel)
-
-        return pruned, {s: s_rel for s, s_rel in relatives.items() if s not in pruned}
-
-
-    def get_fam_samples(s, fam, samples_rel):
-        fam.add(s)
-        for s2 in samples_rel[s]:
+def get_duplicated_samples(
+        kin_ht: hl.Table,
+        i_col: str = 'i',
+        j_col: str = 'j',
+        kin_col: str = 'kin',
+        duplicate_threshold: float = 0.4 # TODO: Look at distribution to pick a value
+) -> Dict[str, Set[str]]: #TODO: Is this the most intuitive format? Otherwise could be List[Set[str]] where each set is a set of duplicated samples.
+    """
+    Given a pc_relate output Table, extract the list of duplicate samples. Returns a dict where the keys are the
+    samples that have one or more duplicates and the values are the set duplicates for each sample.
+
+
+    :param Table kin_ht: pc_relate output table
+    :param str i_col: Column containing the 1st sample
+    :param str j_col: Column containing the 2nd sample
+    :param str kin_col: Column containing the kinship value
+    :param float duplicate_threshold: Kinship threshold to consider two samples duplicated
+    :return: Dict of samples that have are duplicated and their duplicated IDs
+    :rtype: dict of str: set of str
+    """
+
+    dups = kin_ht.filter(kin_ht[kin_col] > duplicate_threshold).collect()
+
+    duplicated_samples = defaultdict(set)
+    for row in dups:
+        if row[kin_col] > duplicate_threshold:
+            duplicated_samples[row[i_col]].add(row[j_col])
+            duplicated_samples[row[j_col]].add(row[i_col])
+
+    return duplicated_samples
+
+
+def infer_families(kin_ht: hl.Table,
+                   sex_ht: hl.Table,
+                   duplicated_samples: Set[str],
+                   i_col: str = 'i',
+                   j_col: str = 'j',
+                   kin_col: str = 'kin',
+                   k2_col: str = 'k2',
+                   sex_s_col: str = 's',
+                   sex_sex_col: str = 'sex',
+                   first_degree_threshold: Tuple[float, float] = (0.23, 0.27),  # TODO: Look at distribution
```

Comment:
I'd probably suggest moving these out of the signature. Once you look at the distribution, these are kinda fixed values, and not something the user should ever really change...

Reply:
I'll keep these here for now to make sure I don't forget about them. I feel like having them here makes sense in terms of clarity, even though I don't think people should be toying with them much.